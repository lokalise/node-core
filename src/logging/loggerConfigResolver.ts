import type { LoggerOptions } from 'pino'

export type AppLoggerConfig = {
  logLevel: 'fatal' | 'error' | 'warn' | 'info' | 'debug' | 'trace' | 'silent'
  nodeEnv: 'production' | 'development' | 'test'
}

export function resolveLoggerConfiguration(appConfig: AppLoggerConfig): LoggerOptions {
  const config: LoggerOptions = {
    level: appConfig.logLevel,
    formatters: {
      level: (label) => {
        return { level: label }
      },
    },
  }
  if (appConfig.nodeEnv !== 'production') {
    config.transport = {
      target: 'pino-pretty',
      options: {
        colorize: true,
        translateTime: 'SYS:standard',
        ignore: 'hostname,pid',
      },
    }
  }
  return config
}
