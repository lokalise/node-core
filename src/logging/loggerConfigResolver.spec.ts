import { pino } from 'pino'

import { resolveLoggerConfiguration } from './loggerConfigResolver'

describe('loggerConfigResolver', () => {
  describe('resolveLoggerConfiguration', () => {
    it('resolves prod configuration', () => {
      const loggerConfig = resolveLoggerConfiguration({
        logLevel: 'warn',
        nodeEnv: 'production',
      })

      expect(loggerConfig).toMatchSnapshot()
    })

    it('resolves dev configuration', () => {
      const loggerConfig = resolveLoggerConfiguration({
        logLevel: 'debug',
        nodeEnv: 'development',
      })

      expect(loggerConfig).toMatchSnapshot()
    })

    it('does not crash during label resolution', () => {
      const loggerConfig = resolveLoggerConfiguration({
        logLevel: 'warn',
        nodeEnv: 'production',
      })

      const logger = pino(loggerConfig)

      logger.warn('test')
    })
  })
})
