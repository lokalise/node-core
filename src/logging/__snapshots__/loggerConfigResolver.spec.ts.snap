// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loggerConfigResolver resolveLoggerConfiguration resolves dev configuration 1`] = `
{
  "formatters": {
    "level": [Function],
  },
  "level": "debug",
  "transport": {
    "options": {
      "colorize": true,
      "ignore": "hostname,pid",
      "translateTime": "SYS:standard",
    },
    "target": "pino-pretty",
  },
}
`;

exports[`loggerConfigResolver resolveLoggerConfiguration resolves prod configuration 1`] = `
{
  "formatters": {
    "level": [Function],
  },
  "level": "warn",
}
`;
